File: Java/repl/REPL.java
package repl;

import java.util.Scanner;
import lexer.Lexer;
import token.Token;
public class REPL {

    private static final String PROMPT = ">> ";

    public static void start() {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print(PROMPT);

            if (!scanner.hasNextLine()) {
                scanner.close();
                return;
            }

            String line = scanner.nextLine();

            Lexer lexer = new Lexer(line);
            while (true) {
                Token x = lexer.nextToken();
                if (x.type.equals("EOF")) break;
                if (x.type.equals("ILLEGAL")) {
                    System.out.println("ILLEGAL token at: " + x.literal);
                    break;
                } else {
                    System.out.println(x.type + " " + x.literal);
                }
            }
        }
    }


    public static void main(String[] args) {
        start(); 
    }
}

File: Aviral/myrepl/repl.py
# repl/repl.py
from Aviral.mylexer.lexer import Lexer
from Aviral.mytoken.token import Token
from Aviral.mytoken.tokentype import TokenType
PROMPT = ">> "

def start():
    while True:
        try:
            line = input(PROMPT)
        except EOFError:
            return

        lexer = Lexer(line)

        while True:
            tok = lexer.next_token()
            if tok.type == "EOF":
                break
            if tok.type == "ILLEGAL":
                print(f"ILLEGAL token at: {tok.literal}")
                break
            else:
                print(f"{tok.type} {tok.literal}")

